test/registry/TestnetAddressRegistry.t.sol

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {Test} from "forge-std/Test.sol";
+import {AddressRegistry} from "src/improvements/AddressRegistry.sol";
+
+contract TestnetAddressRegistryTest is Test {
+    AddressRegistry private addresses;
+
+    uint256 public opSepoliaChainId;
+    uint256 public metalSepoliaChainId;
+
+    function setUp() public {
+        string memory networkConfigFilePath = "test/tasks/mock/configs/DiscoverChainAddressesTestnetConfig.toml";
+
+        vm.createSelectFork("sepolia");
+
+        addresses = new AddressRegistry(networkConfigFilePath);
+        opSepoliaChainId = getChain("optimism_sepolia").chainId;
+        metalSepoliaChainId = getChain("metal_sepolia").chainId;
+    }
+
+    function testContractState() public view {
+        assertTrue(addresses.supportedL2ChainIds(opSepoliaChainId), "Op Sepolia chain ID not supported");
+    }
+
+    function testSuperchainAddressesLoaded() public view {
```

Comment:
It feels like we should be able to abstract this test suite better instead of duplicating the code. For example an `AddressRegistryBaseTest` contract containing all the code, then `AddressRegistry{Mainnet|Sepolia}Test is AddressRegistryBaseTest` where `setUp` calls `super.setUp` and adds ` vm.createSelectFork("mainnet|sepolia")`. 

Alternatively,  a single test file where the tests themselves are internal methods and take the `addresses` contract to use as an input. In `setUp` we define the mainnet and sepolia address contracts and a test looks like:

```solidity
function testSuperchainAddressesLoaded() public view {
  _testSuperchainAddressesLoaded("mainnet");
  _testSuperchainAddressesLoaded("sepolia");
}
```